kodekloudhub/amazon-elastic-kubernetes-service-course
# Unify eks

This should be merged at the same time as https://github.com/kodekloudhub/certified-kubernetes-administrator-course/pull/177

The aim of this PR is to unify the instructions for building an EKS cluster with terraform in one place, thus applying the policy *single source of truth*.

To date, we have had these instructions initially in the CKA repo and then copied to here, creating a maintenance issue in that if EKS changes (which it recently has), then the edits need to be made in two places.

Changes in this PR
* Update for latest EKS
    * Add the `access_config` block to `aws_eks_cluster` resource
    * Update the node image AMI (Kube 1.31)
* Mitigation for environment issue where sometimes not all subnets are created in VPC when lab launches (should be investigated by Sai).
* Fix deprecated `managed_policy_arns` in `aws_iam_role` resource
* Update documentation markdown to give landing pages for CKA and AWS EKS builds.

The markdown has been edited in line with what is currently in the [course](https://learn.kodekloud.com/user/courses/aws-eks/module/fdf5f38f-ffcc-4b70-bde9-751c06d39ac1/lesson/25419f99-dffc-4624-93e8-253cf9703618) question panes for Q3 and Q4. In the interest of not having to modify the course when this repo is changed, the following changes should be made to the course
* Q3 - Link to https://github.com/kodekloudhub/amazon-elastic-kubernetes-service-course/docs/deploy.md and inform student to follow instructions there.
* Q4 - Link to https://github.com/kodekloudhub/amazon-elastic-kubernetes-service-course/docs/nodes.md and inform student to follow instructions there.

Obviously these links won't be correct until after this PR is merged!

This build has been tested against the EKS lab although there are [pre-existing bugs](https://app.clickup.com/t/86cxc2nzu) in the lab.
